### DISCUSSION TITLE
Backblaze buckets often do not load

### DISCUSSION THREAD

**Discussion by Jstaud** [iterate-ch/cyberduck#14713] [Q&A]
When opening a connection to backblaze b2, it will often not load the bucket list. It works sometimes and sometimes it doesn't. 
![Screenshot 2023-04-12 at 10 02 48 AM](https://user-images.githubusercontent.com/3604900/231499887-345f449c-f576-41fb-9daa-e497fd74c6e5.png)

**Comment by metadaddy** [comment]
Hi @Jstaud - when you create an application key that has access to only a single bucket, you can optionally enable the "list all bucket names" permission.

<img src="https://user-images.githubusercontent.com/723517/231826319-187f3fd7-141e-4fd3-a488-c460437d9bc9.png" width=50% height=50%>

Does the key you're using have this permission? You can check in the Backblaze web console, or using the B2 CLI:

![image](https://user-images.githubusercontent.com/723517/231825600-8b23e6a3-7c5e-4e1e-9695-6a456d0b3e43.png)

```
% b2 list-keys --long | grep metadaddy-private | awk '{ print $2, $7 }'
metadaddy-private listAllBucketNames,listBuckets,readBuckets,listFiles,readFiles,shareFiles,writeFiles,deleteFiles,readBucketEncryption,writeBucketEncryption
```

If you can confirm that this is happening with keys that have `listAllBucketNames`, then I'll be happy to take a look.

In any case, it might be worth adding code to report to the user that the key does not have the `listAllBucketNames` permission, to avoid the confusion. What do you think, @dkocher? 

Thanks!

**Comment by Jstaud** [comment]
Hi there,I can confirm that this key was generated with the option to “list all bucket names” permission enabled. It does list all buckets sometimes, but not consistently. I have regenerated new keys to ensure that it’s not just an anomaly with a specific key in Backblaze. Thank you!

**Comment by metadaddy** [comment]
Thanks, @Jstaud - I'll see if I can reproduce. What version of Cyberduck are you using? Have you used other versions in the past without an issue?

**Reply by Jstaud** [reply]
I’ll check the version this morning and reply and I’ll check again to see if this is still an issue.

**Reply by Jstaud** [reply]
Version 8.5.9 (39636)

**Reply by Jstaud** [reply]
And I think previous versions had the same issue but not as frequent. I just replicated the issue today.

**Comment by dkocher** [comment]
@metadaddy When connecting with an application key that does  not have the `listAllBucketNames` permission, only the single bucket will be displayed [^1].

[^1]: 894860a7

**Comment by dkocher** [comment]
It does *not* look like a duplicate of #14527 as this is related to listing buckets and not its contents.

**Comment by Jstaud** [comment]
This is still an issue and has now rendered the application unusable for my team.
